lib/puppet/module.rb

```diff
@@ -90,7 +90,7 @@ def has_metadata?
     return false unless Puppet::FileSystem.exist?(metadata_file)
 
     begin
-      metadata =  JSON.parse(File.read(metadata_file))
+      metadata =  JSON.parse(File.read(metadata_file, :encoding => 'utf-8'))
```

Comment:
This makes me nervous. The metadata file is originally constructed by whomever published the module, eg on windows. So that file could be encoded using the windows active code page. The user could also have published the module awhile ago, e.g. using puppet 3.7, using PMT that didn't enforce utf-8 encoding when writing the metadata.

The user install it on a *nix master using puppet 4.5. The master then tries to load the module and parse the metadata, and it raises an exception because the metadata contents are not utf-8 encoded.

Do we specify somewhere that module metadata must be utf-8 encoded? Should we fall back to the old behavior if a specific encoding-related exception is raised?